gcalbot/gcalbot/http.go

```diff
@@ -181,6 +198,26 @@ func (h *HTTPSrv) configHandler(w http.ResponseWriter, r *http.Request) {
 	// if the calendar hasn't changed, update the settings
 	if calendarID == previousCalendarID {
 		h.Stats.Count("config - update")
+
+		if (!page.Invite && page.Reminder == "") && (inviteInput != "" || reminderInput != "") {
+			// this update must open a new webhook channel, do that now and if it errors, fail early
+			err = h.handler.createEventChannel(selectedAccount, calendarID)
+			if err != nil {
+				switch typedErr := err.(type) {
+				case *googleapi.Error:
+					for _, errorItem := range typedErr.Errors {
+						if errorItem.Reason == "pushNotSupportedForRequestedResource" {
+							page.PushNotAllowed = true
+							h.servePage(w, "config", page)
+							err = nil // clear error
+							return
+						}
+					}
+				}
```

Comment:
Errorf here?

Reply:
This is a kind of unavoidable error that Google may or may not through when subscribing to a calendar (for example trying to subscribe to push notifications for your Contacts calendar seems to fail - I could probably hard code hiding the Contacts calendar in this case but there could be others) so I wanted to display that error in a readable way to the user. I could do a Debug, but I think an Errorf might be overkill because it's not an "us" issue - it's Google

Reply:
I thought we caught the exception for the subscription failure, i just wanted to log any others so we can be aware